drivers/imu/adis.c

```diff
@@ -2532,29 +2528,49 @@ int adis_read_burst_data(struct adis_dev *adis, uint8_t buff_size,
 
 	adis->diag_flags.checksum_err = false;
 
-	if (burst32 && !(adis->info->flags & ADIS_HAS_BURST32)) {
-		/* burst32 requested, but device does not support burst32 readings,
-		thus reading directly from registers */
-		uint32_t reg_val;
-		/* Diag data */
-		buff[0] = buffer[ADIS_READ_BURST_DATA_CMD_SIZE];
-
-		for (idx = 0; idx < 6; idx++) {
-			buff[idx * 2] = buffer[idx + ADIS_READ_BURST_DATA_CMD_SIZE + 1];
-			ret = adis_read_reg(adis, adis->info->field_map->x_gyro.reg_addr + idx * 4,
-					    &reg_val, 2);
-			if (ret)
-				return ret;
-			buff[idx * 2 + 1] = reg_val;
-		}
-		/* Temp data */
-		buff[13] = buffer[7];
-		/* Counter data */
-		buff[14] = buffer[8];
+	uint8_t axis_data_size = 12;
+	if (burst32)
+		axis_data_size = 24;
+
+	uint8_t axis_data_offset = ADIS_READ_BURST_DATA_CMD_SIZE + 2;
+	uint8_t temp_offset = axis_data_offset + axis_data_size;
+	uint8_t data_cntr_offset = temp_offset + 2;
+
+	if (burst32) {
+		memcpy(&data->x_gyro_lsb, &buffer[axis_data_offset], 2);
```

Comment:
Can't we just have 32 bit variables for each of these values and just arrange the two 16 bit parts? It would simplify things a bit if the `adis_burst_data` struct is meant to be used directly in an application.

Reply:
I would say no. if the user is interested only in 16 bit burst reads, then he should only be concerned about the msb part and only handle those values. furthermore, there are devices which support only 16-bit burst reads so this structure would be useful if support is needed in the future.